src/Core/Format/FormatLoader.php

```diff
@@ -0,0 +1,62 @@
+<?php namespace Qosasa\Core\Format;
+
+use Qosasa\Core\Exceptions\FormatProviderResolverException;
+use Qosasa\Core\Format\FormatInflater;
+use Qosasa\Core\Format\Providers\JSONFormatProvider;
+use Qosasa\Core\Snippet;
+
+
+class FormatLoader {
+
+    /**
+     * Create a new format loader
+     *
+     * @param  \Qosasa\Core\Snippet  $snippet
+     * @return void
+     */
+    public function __construct(Snippet $snippet, FormatInflater $formatInflater)
+    {
+        $this->snippet = $snippet;
+        $this->formatInflater = $formatInflater;
+    }
+
+    /**
+     * Load the format
+     *
+     * @return Format
+     */
+    public function load()
+    {
+        // Get info from snippet
+        $file = $this->snippet->getFormatFile();
+        $providerName = $this->snippet->getProvider();
```

Comment:
I'll switch to getFormatProvider, but let's keep that responsibility to the snippet to know what's the $providerName.